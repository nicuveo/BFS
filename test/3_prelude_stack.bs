include "common/check"

def test_dupc(C c) {
  pushc(c)
  dupc check("dupc")
  printc_ord endl popc
  printc_ord endl popc
}

def test_dupi(I i) {
  pushi(i)
  dupi check("dupi")
  printi endl popi
  printi endl popi
}

def test_swapc(C c1, C c2) {
  pushc(c1) pushc(c2)
  swapc check("swapc")
  printc_ord endl popc
  printc_ord endl popc
}

def test_swapi(I i1, I i2) {
  pushi(i1) pushi(i2)
  swapi check("swapi")
  printi endl popi
  printi endl popi
}

def main() {
  test_dupc(0)
  test_dupc(1)
  test_dupc('*')

  test_dupi(0)
  test_dupi(1)
  test_dupi(256)
  test_dupi(65536)

  test_swapc(1,2)
  test_swapi(1,2)
}
