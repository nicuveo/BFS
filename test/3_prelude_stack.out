0
0
1
1
42
42
0
0
1
1
256
256
65536
65536
1
2
1
2
